pkg/plugin/integration_test.go

```diff
@@ -351,27 +315,256 @@ func TestCreateThenUpdateCausesNoRotations(t *testing.T) {
+
+	tests := []struct {
+		name                  string
+		change                func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error
+		expectMasterRotation  bool
+		expectSyncRotation    bool
+		expectComputeRotation bool
+		expectCalls           []string
+	}{
+		{
+			name:                  "update with no changes",
+			expectMasterRotation:  false,
+			expectSyncRotation:    false,
+			expectComputeRotation: false,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "scale up",
+			expectMasterRotation:  false,
+			expectSyncRotation:    false,
+			expectComputeRotation: false,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				for _, p := range cs.Properties.AgentPoolProfiles {
+					if p.Role == api.AgentPoolProfileRoleCompute {
+						p.Count = 4
+					}
+				}
+				perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "change vm image",
+			expectMasterRotation:  true,
+			expectSyncRotation:    true,
+			expectComputeRotation: true,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				cs.Config.ImageVersion = "311.12.12345678"
+				perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "change controller loglevel",
+			expectMasterRotation:  true,
+			expectSyncRotation:    false,
+			expectComputeRotation: false,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				cs.Config.ComponentLogLevel.ControllerManager = to.IntPtr(5)
+				perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "change container image",
+			expectMasterRotation:  false,
+			expectSyncRotation:    true,
+			expectComputeRotation: false,
+			change: func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error {
+				cs.Config.Images.WebConsole = "newImage"
+				perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+				if perr != nil {
+					return perr
+				}
+				return nil
+			},
+		},
+		{
+			name:                  "rotate cluster secrets",
+			expectMasterRotation:  true,
+			expectSyncRotation:    true,
+			expectComputeRotation: false,
```

Comment:
I would expect that rotating cluster secrets would also cause compute vm rotations

Reply:
I'll double check the way i am getting the update blobs, but might be good to look at the e2e test for cluster secrets. what's the command to run it again?

Reply:
@asalkeld you can run the cluster secrets rotation with 
```
FOCUS="\[KeyRotation\]\[Fake\]" ./hack/e2e.sh
```